ocs_ci/utility/kubevirt_vm.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf8 -*-
+"""
+Module for interactions with IBM Cloud Cluster.
+
+"""
+
+import logging
+import time
+
+from ocs_ci.deployment.helpers.hypershift_base import get_cluster_vm_namespace
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility.decorators import switch_to_provider_for_function
+from ocs_ci.ocs import constants
+
+
+logger = logging.getLogger(name=__file__)
+
+
+def get_vm_name(vm):
+    """
+    Get the virtual machine name
+
+    Args:
+        vm (dict): The dictionary that represents the virtual machine
+
+    Returns:
+        str: The virtual machine name
+
+    """
+    return vm["metadata"]["name"]
+
+
+class KubevirtVM(object):
+    """
+    Wrapper for VM objects with kubevirt. The class should be used in Provider mode
+    when we have provider and client clusters in the run, and the client is the primary cluster.
+    """
+
+    def __init__(self):
+        """
+        Constructor for access and modify the virtual machine (VM) in the cluster.
+        The class should be used in Provider mode when we have provider and client clusters in the run,
+        and the client is the primary cluster.
+
+        """
+        self.vm_namespace = get_cluster_vm_namespace()
+        provider_index = config.get_provider_index()
+        self.vm_kubeconfig = config.clusters[provider_index].RUN["kubeconfig"]
+
+        self.ocp_vm = OCP(
+            kind="vm",
+            cluster_kubeconfig=self.vm_kubeconfig,
+            namespace=self.vm_namespace,
+        )
+
+    @switch_to_provider_for_function
+    def run_kubevirt_vm_cmd(
+        self,
+        cmd,
+        secrets=None,
+        timeout=600,
+        ignore_error=False,
+        **kwargs,
+    ):
+        """
+        The wrapper function for 'run_cmd' that runs the VM commands.
+
+        Args:
+            cmd (str): command to run
+            secrets (list): A list of secrets to be masked with asterisks
+                This kwarg is popped in order to not interfere with
+                subprocess.run(``**kwargs``)
+            timeout (int): Timeout for the command, defaults to 600 seconds.
+            ignore_error (bool): True if ignore non zero return code and do not
+                raise the exception.
+
+        """
+        cmd = f"virtctl {cmd} -n {self.vm_namespace}"
+        return run_cmd(cmd, secrets, timeout, ignore_error, **kwargs)
```

Comment:
run_cmd is a deprecated form of exec_cmd, do we really need it here?

Reply:
Ack. I will check if we can change it.